packages/zowe-explorer/src/ZoweExplorerExtender.ts

```diff
@@ -191,14 +191,51 @@ export class ZoweExplorerExtender implements ZoweExplorerApi.IApiExplorerExtende
                 });
             }
         }
-        if (profileTypeConfigurations !== undefined) Profiles.getInstance().addToConfigArray(profileTypeConfigurations);
+
+        if (hasProfileTypes) {
+            Profiles.getInstance().addToConfigArray(profileTypeConfigurations);
+        }
+
+        // Check if schema needs updated when the end user is using a team config
+        if (usingTeamConfig) {
+            this.updateSchema(profileInfo, profileTypeConfigurations);
+        }
 
         // sequentially reload the internal profiles cache to satisfy all the newly added profile types
         await ZoweExplorerExtender.refreshProfilesQueue.add(async (): Promise<void> => {
             await Profiles.getInstance().refresh(ZoweExplorerApiRegister.getInstance());
         });
     }
 
+    /**
+     * Adds new types to the Zowe schema.
+     * @param profileInfo a ProfileInfo object that has been prepared with `readProfilesFromDisk` (such as the one initialized in `initForZowe`).
+     * @param profileTypeConfigurations (optional) Profile type configurations to add to the schema
+     */
+    private updateSchema(
+        profileInfo: zowe.imperative.ProfileInfo,
+        profileTypeConfigurations?: zowe.imperative.ICommandProfileTypeConfiguration[]
+    ): void {
+        if (profileTypeConfigurations) {
+            try {
+                for (const typeConfig of profileTypeConfigurations) {
+                    const addResult = profileInfo.addProfileTypeToSchema(typeConfig.type, {
+                        schema: typeConfig.schema,
+                        sourceApp: "Zowe Explorer (for VS Code)",
+                    });
+                    if (addResult.info.length > 0) {
+                        Gui.warningMessage(addResult.info);
+                    }
+                }
+            } catch (err) {
+                // Only show an error if we failed to update the on-disk schema.
+                if (err.code === "EACCES" || err.code === "EPERM") {
+                    Gui.errorMessage(localize("schema.cannotAccess", "Failed to update Zowe schema: insufficient permissions or read-only file"));
```

Comment:
Would it be good to append the original error message at the end of ours, to offer more error details to users?

Reply:
Yes - I've added the error details to the end of the message if the `message` property is present.